pycbc/psd/variation.py

```diff
@@ -160,3 +165,213 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
     ind -= 1
     vals = psd_var[ind]
     return vals
+
+
+def mean_square(data, delta_t, short_stride, stride):
+    """ Calculate mean square of given time series once per stride
+
+    First of all this function calculate the mean square of given time
+    series once per short_stride. This is used to remove outliers due
+    to short glitches. Then, every seconds it computes the mean square
+    of the smoothed time series, within the stride.
+
+    Parameters
+    ----------
+    data : TimeSeries
+    delta_t : float
+        Duration of the time series
+    short_stride : float
+        Stride duration for outlier removal
+    stride ; float
+        Stride duration
+
+    Returns
+    -------
+    m_s: List
+        Mean square of given time series
+    """
+
+    srate = int(data.sample_rate)
+
+    # Calculate mean square of data once per short stride and remove
+    # the outliers
+    data = numpy.array(data)
+    short_ms = numpy.mean(data.reshape(-1, int(srate * short_stride)) ** 2,
+                          axis=1)
+    ave = 0.5 * (short_ms[2:] + short_ms[:-2])
+    outliers = short_ms[1:-1] > (2 * ave)
+    short_ms[1:-1][outliers] = ave[outliers]
+
+    # Calculate mean square of data every step with a window equal to
+    # stride seconds
+    m_s = []
+    inv_time = int(1. / short_stride)
+    for index in range(int(delta_t - stride + 1)):
+        m_s.append(numpy.mean(short_ms[inv_time * index:inv_time *
+                                       int(index+stride)]))
+    return m_s
+
+
+def calc_filt_psd_variation(strain, segment, short_segment, psd_long_segment,
+                            psd_duration, psd_stride, psd_avg_method, low_freq,
+                            high_freq):
+    """ Calculates time series of PSD variability
+
+    This function first splits the segment up into 512 second chunks. It
+    then calculates the PSD over this 512 second. The PSD is used to
+    to create a filter that is the composition of three filters:
+    1. Bandpass filter between f_low and f_high.
+    2. Weighting filter which gives the rough response of a CBC template.
+    3. Whitening filter.
+    Next it makes the convolution of this filter with the stretch of data.
+    This new time series is given to the "mean_square" function, which
+    computes the mean square of the timeseries within an 8 seconds window,
+    once per second.
```

Comment:
does the mean_square actually do a calculation once per second?  I guess the assignment of 'times_long =' does do that, so in a way that is hard coding the stride of that calculation.  
Maybe it would be preferable to allow the calculation to be done with a different stride, ie have a `mean_square_stride` argument which in most cases will be 1 but could be some other number?

Reply:
Yes, mean_square do the calculation once per second, and it is not related to times_long. It does it in this loop:

for index in range(int(delta_t - stride + 1)):
        m_s.append(numpy.mean(short_ms[inv_time * index:inv_time *
                                       int(index+stride)]))

 I defined "index" running over "range(int(delta_t - stride + 1))" without explicitly set the step (that is 1 therefore). 

Anyway mean_square compute the mean_square over an 8 second stride, and it compute the output every second.

Reply:
oh, I see. I have some more comments on the arguments given to mean_square & will consider this point along with them.